pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -493,44 +493,27 @@ func manageOpenShiftControllerManagerDeployment_v311_00_to_latest(
 		}
 	} else {
 		for i, c := range required.Spec.Template.Spec.Containers {
-			newEnvs := []corev1.EnvVar{}
-
-			if len(c.Env) == 0 {
-				if len(proxyCfg.Status.NoProxy) > 0 {
-					newEnvs = append(newEnvs, corev1.EnvVar{Name: "NO_PROXY", Value: proxyCfg.Status.NoProxy})
-				}
-				if len(proxyCfg.Status.HTTPProxy) > 0 {
-					newEnvs = append(newEnvs, corev1.EnvVar{Name: "HTTP_PROXY", Value: proxyCfg.Status.HTTPProxy})
-				}
-				if len(proxyCfg.Status.HTTPSProxy) > 0 {
-					newEnvs = append(newEnvs, corev1.EnvVar{Name: "HTTPS_PROXY", Value: proxyCfg.Status.HTTPSProxy})
+			// Remove the relevant variables from the spec.
+			newEnvs := make([]corev1.EnvVar, 0, len(c.Env))
+			for _, env := range c.Env {
+				switch strings.TrimSpace(env.Name) {
+				case "HTTPS_PROXY", "HTTP_PROXY", "NO_PROXY":
```

Comment:
basically, we just want to try to hit this branch of removing the old env variable and see that we get the new value